compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -265,21 +267,38 @@ impl<'a> FunctionContext<'a> {
     /// otherwise values like 2^128 can be assigned to a u8 without error or wrapping.
     pub(super) fn checked_numeric_constant(
         &mut self,
-        value: impl Into<FieldElement>,
+        value: &BigInt,
         typ: Type,
     ) -> Result<ValueId, RuntimeError> {
-        let value = value.into();
+        let negative = value.is_negative();
+        let field = noirc_frontend::utils::big_int_to_field_element(value);
 
-        if let Type::Numeric(typ) = typ {
-            if !typ.value_is_within_limits(value) {
+        if let Type::Numeric(numeric_type) = typ {
+            if !numeric_type.value_is_within_limits(value) {
                 let call_stack = self.builder.get_call_stack();
-                return Err(RuntimeError::IntegerOutOfBounds { value, typ, call_stack });
+                return Err(RuntimeError::IntegerOutOfBounds {
+                    value: value.clone(),
```

Comment:
I think this clone is okay: after this the program will exit, so at most we allocate once.

Reply:
Yep, we clone in error messages pretty frequently.